applications/ShallowWaterApplication/custom_utilities/move_shallow_water_particle_utility.h

```diff
@@ -0,0 +1,2288 @@
+        #else
+            int number_of_threads = 1;
+        #endif
+        OpenMPUtils::CreatePartition(number_of_threads, mrModelPart.Nodes().size(), node_partition);
+        
+        #pragma omp parallel for
+        for(int kkk=0; kkk<number_of_threads; kkk++)
+        {
+            for(unsigned int ii=node_partition[kkk]; ii<node_partition[kkk+1]; ii++)
+            {
+                ModelPart::NodesContainerType::iterator inode = inodebegin+ii;
+                inode->FastGetSolutionStepValue(DELTA_SCALAR1) = inode->FastGetSolutionStepValue(mScalarVar1) - inode->FastGetSolutionStepValue(PROJECTED_SCALAR1);  //PROJECTED_SCALAR1
+                inode->FastGetSolutionStepValue(DELTA_VECTOR1) = inode->FastGetSolutionStepValue(mVectorVar1) - inode->FastGetSolutionStepValue(PROJECTED_VECTOR1);  //PROJECTED_VECTOR1
+            }
+        }
+        KRATOS_CATCH("")
+    }
+
+
+    void CopyScalarVarToPreviousTimeStep(const Variable<double>& OriginVariable,
+                   ModelPart::NodesContainerType& rNodes)
+    {
+        KRATOS_TRY
+        ModelPart::NodesContainerType::iterator inodebegin = rNodes.begin();
+        vector<unsigned int> node_partition;
+        #ifdef _OPENMP
+            int number_of_threads = omp_get_max_threads();
+        #else
+            int number_of_threads = 1;
+        #endif
+        OpenMPUtils::CreatePartition(number_of_threads, rNodes.size(), node_partition);
+        
+        #pragma omp parallel for
+        for(int kkk=0; kkk<number_of_threads; kkk++)
+        {
+            for(unsigned int ii=node_partition[kkk]; ii<node_partition[kkk+1]; ii++)
+            {
+                ModelPart::NodesContainerType::iterator inode = inodebegin+ii;
+                inode->GetSolutionStepValue(OriginVariable,1) = inode->FastGetSolutionStepValue(OriginVariable);
+            }
+        }
+        KRATOS_CATCH("")
+    }
+
+
+    void CopyVectorVarToPreviousTimeStep(const Variable<array_1d<double,3>>& OriginVariable,
+                   ModelPart::NodesContainerType& rNodes)
+    {
+        KRATOS_TRY
+        ModelPart::NodesContainerType::iterator inodebegin = rNodes.begin();
+        vector<unsigned int> node_partition;
+        #ifdef _OPENMP
+            int number_of_threads = omp_get_max_threads();
+        #else
+            int number_of_threads = 1;
+        #endif
+        OpenMPUtils::CreatePartition(number_of_threads, rNodes.size(), node_partition);
+        
+        #pragma omp parallel for
+        for(int kkk=0; kkk<number_of_threads; kkk++)
+        {
+            for(unsigned int ii=node_partition[kkk]; ii<node_partition[kkk+1]; ii++)
+            {
+                ModelPart::NodesContainerType::iterator inode = inodebegin+ii;
+                noalias(inode->GetSolutionStepValue(OriginVariable,1)) = inode->FastGetSolutionStepValue(OriginVariable);
+            }
+        }
+        KRATOS_CATCH("")
+    }
+
+
+    //~ void ComputeVelocity(ModelPart::NodesContainerType& rNodes)
```

Comment:
I'd recommend removing commented out functions in the future (the idea is that you can always access the git history to see old versions of the code)